app/src/main/java/org/mozilla/fenix/components/history/PagedHistoryProvider.kt

```diff
@@ -27,18 +30,21 @@ sealed class HistoryDB {
     abstract val title: String
     abstract val visitedAt: Long
     abstract val selected: Boolean
+    abstract val historyTimeGroup: HistoryItemTimeGroup
 
     data class Regular(
         override val title: String,
         val url: String,
         override val visitedAt: Long,
+        override val historyTimeGroup: HistoryItemTimeGroup,
```

Comment:
I was wondering, aren't we always using `visitedAt` value to calculate `HistoryItemTimeGroup`? If so, wouldn't something like that
```
sealed class HistoryDB {
    abstract val title: String
    abstract val visitedAt: Long
    abstract val selected: Boolean
    val historyTimeGroup = HistoryItemTimeGroup.timeGroupForTimestamp(visitedAt)
```
help to remove some code? Because it looks like we are using one parameter to calculate two, and passing the same parameter twice seems like an overkill. Or I am missing something?

Reply:
Yes, I had the same idea during development.  Maybe I was overthinking this.  I was worried that if we add this default behaviour, we end up not copy the the `historyTimeGroup` (if already exists) and do more operations then needed.

Reply:
As agreed in a offline discussion, I will add the default helper function.